src/expression.py

```diff
@@ -0,0 +1,118 @@
+from typing import Any
+from .token_type import TokenType, Token
+
+
+class Expr:
+    def __call__(self) -> Any:
+        raise RuntimeError("Empty Expresson")
+
+
+class Binary(Expr):
+    left: Expr
+    operator: Token
+    right: Expr
+
+    def __init__(self, left: Expr, operator: Token, right: Expr):
+        self.left = left
+        self.operator = operator
+        self.right = right
+
+    def __str__(self) -> str:
+        return f"( {self.operator} {str(self.left)}, {str(self.right)} )"
+
+    def __call__(self) -> Any:
+        if self.operator.token_type == TokenType.STAR:
+            return self.left() * self.right()
+        elif self.operator.token_type == TokenType.SLASH:
+            return self.left() / self.right()
+        elif self.operator.token_type == TokenType.PLUS:
+            left = self.left()
+            right = self.right()
+            state = type(left) == type(right)
+
+            return left + right if state else str(left) + str(right)
+        elif self.operator.token_type == TokenType.MINUS:
+            return self.left() - self.right()
+        elif self.operator.token_type == TokenType.GREATER_EQUAL:
+            return self.left() >= self.right()
+        elif self.operator.token_type == TokenType.GREATER:
+            return self.left() > self.right()
+        elif self.operator.token_type == TokenType.LESSER_EQUAL:
+            return self.left() <= self.right()
+        elif self.operator.token_type == TokenType.LESSER:
+            return self.left() < self.right()
+        elif self.operator.token_type == TokenType.EQUAL_EQUAL:
+            return self.left() == self.right()
+        elif self.operator.token_type == TokenType.EQUAL:
+            raise RuntimeError("Not implemented.")
```

Comment:
We could use NotImplementedError instead

Reply:
Its gonna be changed in a week so why bother?